Lib/difflib.py

```diff
@@ -1054,6 +1052,8 @@ def IS_LINE_JUNK(line, pat=re.compile(r"\s*(?:#\s*)?$").match):
     False
     """
 
+    if pat is None:
+        return line.strip() in {'', '#'}
```

Comment:
Apologies, I didn't notice Tim's earlier comment. 
```suggestion
    if pat is None:
        # Default: match '#' or the empty string
        return line.strip() in '#'
   # Previous versions used the undocumented parameter 'pat' as a
   # match function. Retain this behaviour for compatibility.
```